packages/atlas/src/components/_video/Settings/Settings.tsx

```diff
@@ -0,0 +1,132 @@
+import { FC, useState } from 'react'
+
+import { Text } from '@/components/Text'
+import { SvgActionChevronL, SvgActionChevronR } from '@/components/_icons'
+
+import {
+  GridWrapper,
+  SettingsHeader,
+  SettingsListItemWrapper,
+  SettingsUnorderedList,
+  SettingsWrapper,
+  StyledSvgActionCheck,
+} from './Settings.styles'
+
+export type SettingValue = string | number | boolean
+
+type SettingsListItemProps = {
+  label: string
+  value: SettingValue
+  isOption?: boolean
+  checked?: boolean
+  toggleable?: boolean
+  onSettingClick?: (setting: { value: SettingValue; label: string }) => void
+}
+
+export type SettingsProps = {
+  settings: Setting[]
+}
+
+export type Setting = {
+  options: SettingsListItemProps[]
+} & SettingsListItemProps
+
+export const Settings: FC<SettingsProps> = ({ settings }) => {
+  const [openedOption, setOpenedOption] = useState<string | null>(null)
+
+  const baseMenu: Setting[] = settings.map((s) => ({
+    ...s,
+    onSettingClick: ({ label }) => setOpenedOption(label),
+  }))
+
+  const selectedOptions = settings?.find((s) => s.label === openedOption)
+
+  const options = selectedOptions
+    ? (selectedOptions.options.map((opt) => ({
+        ...opt,
+        onSettingClick: (setting) => {
+          opt?.onSettingClick?.(setting)
+          setOpenedOption(null)
+        },
+      })) as SettingsListItemProps[])
+    : []
+
+  return (
+    <>
+      {openedOption === null ? (
+        <SettingList title="Settings" settings={baseMenu} />
+      ) : (
+        <SettingList
+          title={openedOption}
+          isOption
+          onHeaderClick={() => {
+            return setOpenedOption(null)
+          }}
+          settings={options}
+        />
+      )}
+    </>
+  )
+}
+
+type SettingsListProps = {
+  title: string
+  settings?: SettingsListItemProps[]
+  onHeaderClick?: () => void
+  isOption?: boolean
+}
+
+export const SettingList: FC<SettingsListProps> = ({ settings, title, onHeaderClick, isOption = false }) => {
+  return (
+    <SettingsWrapper>
+      <SettingsHeader
+        isClickable={isOption}
+        onClick={(e) => {
```

Comment:
Sorry, it's probably just me but I really dislike these meaningless arguments. I may be nitpicking, but I think the code should be as self-describing as possible. If you think I'm exaggerating, let me know 😛, if not can we fix these in all places? 🙏